src/components/menu/provider.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, Dispatch, FC, HTMLAttributes, SetStateAction, useState } from 'react'
+
+interface MenuContextValue {
+  triggerProps: HTMLAttributes<HTMLButtonElement>
+  popoverProps: HTMLAttributes<HTMLDivElement>
+  isOpen: boolean
+  setIsOpen: Dispatch<SetStateAction<boolean>>
+}
+
+const MenuContext = createContext<MenuContextValue | null>(null)
+
+const MenuProvider: FC = ({ children }) => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  const triggerProps = {
+    'aria-haspopup': true,
+    'aria-expanded': isOpen,
+    onClick: () => {
+      setIsOpen((prev) => !prev)
+    },
+  }
+
+  const popoverProps = {
+    role: 'menu',
+    hidden: !isOpen,
```

Comment:
> question: does the use of hidden here make it harder for us to animate the opening/closing of the popover?

no, turns out it doesn't required since we custom the `display` prop via css

>suggestion: If it does, I think a data-open attribute would make sense.

ok, I'll try it